crates/ra_hir/src/source_id.rs

```diff
@@ -5,15 +5,18 @@ use std::{
 };
 
 use ra_arena::{impl_arena_id, Arena, RawId};
-use ra_syntax::{ast, AstNode, SyntaxNode, SyntaxNodePtr};
+use ra_syntax::{
+    ast::{self, AttrsOwner},
+    AstNode, SyntaxNode, SyntaxNodePtr,
+};
 
 use crate::{AstDatabase, HirFileId};
 
 /// `AstId` points to an AST node in any file.
 ///
 /// It is stable across reparses, and can be used as salsa key/value.
 #[derive(Debug)]
-pub(crate) struct AstId<N: AstNode> {
+pub struct AstId<N: AstNode> {
```

Comment:
It's better if this renmains `pub(crate)

Reply:
Produces the following:

```
error[E0446]: crate-visible type `source_id::AstId<ra_syntax::ast::generated::MacroCall>` in public interface
   --> crates/ra_hir/src/ids.rs:215:30
    |
215 |     Macro { def: MacroDefId, ast_id: AstId<ast::MacroCall> },
    |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak crate-visible type
    | 
   ::: crates/ra_hir/src/source_id.rs:19:1
    |
19  | pub(crate) struct AstId<N: AstNode> {
    | ---------- `source_id::AstId<ra_syntax::ast::generated::MacroCall>` declared as crate-visible
```